assets/src/blocks/Spreadsheet/SpreadsheetFrontend.js

```diff
@@ -16,88 +16,75 @@ const placeholderData = {
   ],
 };
 
-export class SpreadsheetFrontend extends Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      loading: null,
-      spreadSheetData: null,
-      searchText: '',
-      sortDirection: 'asc',
-      sortColumnIndex: null,
-    };
-
-    this.onHeaderClick = this.onHeaderClick.bind(this);
-  }
-
-  onHeaderClick(index) {
-    if (index === this.state.sortColumnIndex) {
-      const newDirection = this.state.sortDirection === 'asc' ? 'desc' : 'asc';
-      this.setState({
-        sortDirection: newDirection,
-      });
+export const SpreadsheetFrontend = ({
+  url,
+  color,
+  setInvalidSheetId,
+  className,
+}) => {
+  const [loading, setLoading] = useState(false);
+  const [spreadsheetData, setSpreadsheetData] = useState(null);
+  const [searchText, setSearchText] = useState('');
+  const [sortDirection, setSortDirection] = useState('asc');
+  const [sortColumnIndex, setSortColumnIndex] = useState(null);
+
+  const onHeaderClick = index => {
+    if (index === sortColumnIndex) {
+      const newDirection = sortDirection === 'asc' ? 'desc' : 'asc';
+      setSortDirection(newDirection);
     } else {
-      this.setState({
-        sortColumnIndex: index,
-        sortDirection: 'asc',
-      });
-    }
-  }
-
-  async fetchSheetData(url) {
-    const sheetID = this.extractSheetID(url);
-
-    if (sheetID !== false) {
-      if (this.props.handleErrors) {
-        this.props.handleErrors({invalidSheetId: false});
-      }
-      this.setState({loading: true});
-
-      const args = {
-        sheet_id: sheetID,
-      };
-
-      const baseUrl = document.body.dataset.nro;
-
-      const spreadSheetData = baseUrl ?
-        await fetchJson(`${baseUrl}/wp-json/${addQueryArgs('planet4/v1/get-spreadsheet-data', args)}`) :
-        await apiFetch({path: addQueryArgs('planet4/v1/get-spreadsheet-data', args)});
-
-      this.setState({loading: false, spreadSheetData});
-    } else {
-      if (this.props.handleErrors) {
-        this.props.handleErrors({invalidSheetId: true});
-      }
-      this.setState({loading: false, spreadSheetData: null});
+      setSortColumnIndex(index);
+      setSortDirection('asc');
     }
-  }
+  };
 
-  async componentDidMount() {
-    if (this.props.url !== '') {
-      await this.fetchSheetData(this.props.url);
-    }
-  }
-
-  // eslint-disable-next-line react/no-deprecated
-  async componentWillReceiveProps(nextProps) {
-    if (nextProps.url !== this.props.url) {
-      await this.fetchSheetData(nextProps.url);
-    }
-  }
-
-  extractSheetID(url) {
+  const extractSheetID = urlParam => {
     const googleSheetsPattern = /https:\/\/docs\.google\.com\/spreadsheets\/d\/e\/([\w-]+)/;
-    const matches = url.match(googleSheetsPattern);
+    const matches = urlParam.match(googleSheetsPattern);
     if (matches !== null) {
       return matches[1];
     }
     return false;
-  }
+  };
+
+  useEffect(() => {
+    const fetchSheetData = async () => {
```

Comment:
We can use [IIFE](https://developer.mozilla.org/en-US/docs/Glossary/IIFE) instead
```javascript
(async () => { ... })()
```

Reply:
Updated!